src/libraries/tests.proj

```diff
@@ -284,7 +284,13 @@
   <ItemGroup>
     <ProjectReference Include="$(MSBuildThisFileDirectory)*\tests\**\*.Tests.csproj"
                       Exclude="@(ProjectExclusions)"
-                      Condition="'$(TestAssemblies)' == 'true'" />
+                      Condition="'$(TestAssemblies)' == 'true' and
```

Comment:
This is a very long complicated condition, couldn't you choose to skip the entire build step like the "Send to Helix" part does?  (or do you need the non-test-project build to run here?)

Reply:
Building the functional tests is important on some platforms, as a way to integration test multiple things like the AOT tasks, LDAP tasks, linker, etc.

I could make the conditional even more complex by restricting it to only specific platforms...

Reply:
Another way to express it might be with a property, like:
```xml
<PropertyGroup>
  <IncludeLibraryTests Condition="'$(TestAssemblies)' != 'true'">false</IncludeLibraryTests>
  <IncludeLibraryTests Condition="'$(IncludeLibraryTests)' == '' and '$(ContinuousIntegrationBuild)' != 'true'">true</IncludeLibraryTests>
  <IncludeLibraryTests Condition="'$(IncludeLibraryTests)' == '' and '$(RuntimeFlavor)' != 'Mono'">true</IncludeLibraryTests>

  <IncludeLibraryTests Condition="'$(IncludeLibraryTests)' == '' and
      '$(RuntimeFlavor)' == 'Mono' and
          ('$(librariesContainsChange)' == 'true' or
           '$(monoContainsChange)' == 'true' or
           '$(isFullMatrix)' == 'true')">true</IncludeLibraryTests>
  <IncludeLibraryTests Condition="'$(IncludeLibraryTests)' == ''">false</IncludeLibraryTests>
</PropertyGroup>
```

Reply:
Maybe we can achieve that on the yaml as well? Either way I'm fine, but I do agree this is a very hard to read condition.